SC-088: DNS TXT Record with Persistent Value DCV Method (#608)
cabforum/servercert: docs/BR.md

**Pending Prohibition​​**: The use of a behavior described with this label is highly discouraged, as it is planned to be deprecated and will likely be designated as MUST NOT in the future.

> **Persistent DCV TXT Record:** A DNS TXT record identifying an Applicant in accordance with Section 3.2.2.4.22.

Author: ```suggestion
**Persistent DCV TXT Record:** A DNS TXT record identifying an Applicant in accordance with [Section 3.2.2.4.22](#322422-dns-txt-record-with-persistent-value).
```

**Note**: Once the FQDN has been validated using this method, the CA MAY also issue Certificates for other FQDNs that end with all the Domain Labels of the validated FQDN. This method is suitable for validating Wildcard Domain Names.

##### 3.2.2.4.22 DNS TXT Record with Persistent Value

> Confirming the Applicant's control over a FQDN by confirming the presence of a persistent record identifying the Applicant in a DNS TXT record for an Authorization Domain Name that is prefixed with one DNS TXT Record Persistent DCV Domain Label.

Author: ```suggestion
Confirming the Applicant's control over a FQDN by confirming the presence of a persistent record identifying the Applicant in a DNS TXT record placed at the "`_validation-persist`" label prepended to the Authorization Domain Name being validated.
```

@ChristopherRC - I want to be careful here to make sure the intent remains clear but what do you think about adding the label directly in the method and removing both the definition and appendix modifications?

Author: While I like the direction, the challenge is what the do with the DNS TXT Record Persistent DCV Domain Label definition. Currently it is a defined term that is essentially a pointer back to the awkward Appendix section. So we'd either have to change the definition of the term or make it an undefined one.

The CA MUST confirm the presence of a TXT record whose RDATA value fulfills the following requirements:

> 1. The RDATA value MUST conform to the `issue-value` syntax as defined in RFC 8659, section 4.2; and

Reviewer: nit (for self consistency in these new requirements)

```suggestion
1. The RDATA value MUST conform to the `issue-value` syntax as defined in RFC 8659, Section 4.2; and
```

2. The `issuer-domain-name` value MUST be an Issuer Domain Name disclosed by the CA in Section 4.2 of the CA's Certificate Policy and/or Certification Practices Statement; and
3. The `issue-value` MUST contain an `accounturi` parameter, where the parameter value is a unique URI (as described by RFC 8657, Section 3) identifying the account of the Applicant which requested validation for this FQDN; and

> 4. The `issue-value` MAY contain a `persistUntil` parameter. If present, the parameter value MUST be a base-10 encoded integer representing a UNIX timestamp (the number of seconds since 1970-01-01T00:00:00Z ignoring leap seconds).

Reviewer 2: ```suggestion
4. The `issue-value` MAY contain a `persistUntil` parameter. If present, the parameter value MUST be a base-10 encoded integer representing a UNIX timestamp (the number of seconds since 1970-01-01T00:00:00Z ignoring leap seconds); and
```

5. The `issue-value` MAY contain additional parameters. CAs MUST ignore any unknown parameter keys.

If the `persistUntil` parameter is present, the CA MUST evaluate its value. If the time of the check is after the time specified in the `persistUntil` parameter value, the CA MUST NOT use the record as evidence of the Applicant's control over the FQDN.
For example, the DNS TXT record might look like:
_validation-persist.example.com IN TXT "authority.example; accounturi=https://authority.example/acct/123; persistUntil=1782424856"

> CAs performing validations using this method MUST implement Multi-Perspective Issuance Corroboration as specified in [Section 3.2.2.9](#3229-multi-perspective-issuance-corroboration). To count as corroborating, a Network Perspective MUST observe the same challenge information as the Primary Network Perspective.

Author: https://github.com/slghtr-says/servercert/pull/3/files#diff-e0ac1bd190515a4f2ec09139d395ef6a8c7e9e5b612957c1f5a2dea80c6a6cfeR1173

@ChristopherRC / @ryancdickson  - Found a gap in 3.2.2.9 (MPIC) as well. 

Current: 
```
* a. the presence of the expected 1) Random Value, 2) Request Token, 3) IP Address, or 4) Contact Address, as required by the relied upon validation method specified in Sections 3.2.2.4 and 3.2.2.5; and
```

Suggested Change: 
```
the presence of the expected  challenge information (i.e. token), as required by the relied upon validation method specified in Sections 3.2.2.4 and 3.2.2.5
```

The DNS TXT record MUST be placed on the "`_validation-contactphone`" subdomain of the domain being validated. The entire RDATA value of this TXT record MUST be a valid Global Number as defined in RFC 3966, Section 5.1.4, or it cannot be used.

> ### A.2.3. DNS TXT Record Persistent DCV Domain Label

Reviewer 2: Administrative comment and sorry for missing this earlier! Considering Appendix A is qualified with the description “CAA Contact Tag”, it’s not entirely clear why the "A.2.3. DNS TXT Record Persistent DCV Domain Label” content was added here beyond possibly following the pre-existing pattern used by the other A.2 entries (which also seems  kind of _awkward_ given the Appendix description). 

Should the naming of Appendix A change, or is there a more appropriate location for the A.2 text (including the newly proposed A.2.3. section)?

Author: Great point. The line after ``APPENDIX A – CAA Contact Tag`` is ``These methods allow domain owners to publish **contact information** in DNS for the purpose of validating domain control.`` which scopes the section to contact based DCV methods. It would get tricky to try to expand the scope of this section. 

That said, I can't really think of a good reason to really maintain the definition and appendix entry either since consistency isn't really achieved here either. I proposed an alternative approach in this comment: https://github.com/cabforum/servercert/pull/608/files#r2323549706

Author: I like it! Especially that this language clearly expresses the normative requirement for the format of the label in the method language itself. It resolves my concern from the MPIC section and improves overall readability of the method.

The definition is a tad circular

Author: Thank you! I think that approach does a some things really well.

1. I like that it adds a normative requirement for the Domain Label format directly in the method itself. 
2. It removes the awkward appendix. 
3. It shifts the approach from defining the Label to defining the Record which is arguably more important since locating that Record at the label is helpful for other areas like the MPIC section. Without that additional definition I couldn't find a clean way to add Method 22 support without it getting too wordy or overly broad. 

I think the definition of the Record is a tad circular but I'm not convinced it makes the meaning less clear. 

I will go ahead and draft up a v2 ballot based all of the feedback provided.

Reviewer: Other feedback / alternative suggestions also welcome!
